qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -69,31 +69,32 @@ def run(self, dag):
         if not self.basis:
             LOG.info("Skipping pass because no basis is set")
             return dag
-        decomposer = OneQubitEulerDecomposer(self.basis)
-        runs = dag.collect_runs(self.euler_basis_names[self.basis])
-        runs = _split_runs_on_parameters(runs)
-        for run in runs:
-            if len(run) <= 1:
-                params = run[0].op.params
-                # Remove single identity gates
-                if run[0].op.name in self.euler_basis_names[self.basis] and len(
-                        params) > 0 and np.array_equal(run[0].op.to_matrix(),
-                                                       np.eye(2)):
-                    dag.remove_op_node(run[0])
-                # Don't try to optimize a single 1q gate
-                continue
-            q = QuantumRegister(1, "q")
-            qc = QuantumCircuit(1)
-            for gate in run:
-                qc.append(gate.op, [q[0]], [])
-
-            operator = Operator(qc)
-            new_circ = decomposer(operator)
-            new_dag = circuit_to_dag(new_circ)
-            dag.substitute_node_with_dag(run[0], new_dag)
-            # Delete the other nodes in the run
-            for current_node in run[1:]:
-                dag.remove_op_node(current_node)
+        for basis in self.basis:
+            decomposer = OneQubitEulerDecomposer(basis)
+            runs = dag.collect_1q_runs()
+            runs = _split_runs_on_parameters(runs)
```

Comment:
Can this be moved into the `filter_fn` for `dagcircuit.collect_1q_runs`?

Reply:
Yeah I assume so it should be as simple as adding `and node.op.is_parameterized()` to the function's return

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5431/commits/5d1ff480d50f08a50dee14ef0750fac4961d69d6